src/bluetooth/qbluetoothtransferreply_bluez.cpp

```diff
@@ -204,51 +250,33 @@ QString QBluetoothTransferReplyBluez::errorString() const
     return m_errorStr;
 }
 
-void QBluetoothTransferReplyBluez::Complete(const QDBusObjectPath &in0)
-{
-    Q_UNUSED(in0);
-    m_transfer_path.clear();
-    m_finished = true;
-    m_running = false;
-}
-
-void QBluetoothTransferReplyBluez::Error(const QDBusObjectPath &in0, const QString &in1)
+void QBluetoothTransferReplyBluez::Complete()
 {
-    Q_UNUSED(in0);
-    m_transfer_path.clear();
+    qCDebug(QT_BT_BLUEZ) << "Transfer complete";
+    delete objectTransfer;
+    objectTransfer = NULL;
     m_finished = true;
     m_running = false;
-    m_errorStr = in1;
-    if (in1 == QStringLiteral("Could not open file for sending")) {
-        m_error = QBluetoothTransferReply::FileNotFoundError;
-        m_errorStr = tr("Could not open file for sending");
-    } else if (in1 == QStringLiteral("Operation canceled")) {
-        m_error = QBluetoothTransferReply::UserCanceledTransferError;
-        m_errorStr = tr("Operation canceled");
-    } else {
-        m_error = QBluetoothTransferReply::UnknownError;
-    }
-
     emit finished(this);
 }
 
-void QBluetoothTransferReplyBluez::Progress(const QDBusObjectPath &in0, qulonglong in1)
+void QBluetoothTransferReplyBluez::Error(const QString &in0, const QString &in1)
 {
-    Q_UNUSED(in0);
-    emit transferProgress(in1, m_size);
-}
-
-void QBluetoothTransferReplyBluez::Release()
-{
-    if(m_errorStr.isEmpty())
-        emit finished(this);
+    qCDebug(QT_BT_BLUEZ) << "Transfer error" << in0 << "," << in1;
+    delete objectTransfer;
+    objectTransfer = NULL;
+    fail(QBluetoothTransferReply::UnknownError, in1);
+    emit finished(this);
 }
 
-QString QBluetoothTransferReplyBluez::Request(const QDBusObjectPath &in0)
+void QBluetoothTransferReplyBluez::Progress(const QString &in0, const QDBusVariant &in1)
 {
-    m_transfer_path = in0.path();
+    qCDebug(QT_BT_BLUEZ) << "Property indication for" << in0;
 
-    return QString();
+    if (in0 == QString("Progress")) {
```

Comment:
QLatin1String(). Note, not QStringLiteral() as operator== has a latin1 override.